File: scripts/simple_adder_client.py
#!/usr/bin/env python3
"""
    Simple ROS client for /add_two_ints service
    
    Author: Jari Honkanen
"""

import rospy
from rospy_tutorials.srv import AddTwoInts

if __name__ == "__main__":

    rospy.init_node("simple_adder_client")
    rospy.wait_for_service("/add_two_ints")   # Block until service is advertised

    try:
        add_two_ints = rospy.ServiceProxy("/add_two_ints", AddTwoInts)
        a = 7
        b = 8
        response = add_two_ints(a,b)
        rospy.loginfo("[RESULT] " + str(a) + " + " + str(b) + " = " + str(response.sum))

    except rospy.ServiceException as e:
        rospy.logwarn("[ERROR] Service call failed: " + str(e))

File: my_simple_robot_nodes/src/simple_adder_client.cpp

/*-------------------------------------------------------------------*\
  NAME
    simple_adder_client.cpp

  DESCRIPTION
    Simple ROS Service that adds two integers

  AUTHOR
    Jari Honkanen
\*-------------------------------------------------------------------*/

#include <ros/ros.h>
#include <rospy_tutorials/AddTwoInts.h>


int main(int argc, char **argv)
{
    ros::init(argc, argv, "simple_adder_client");
    ros::NodeHandle nh;

    ros::ServiceClient client = nh.serviceClient<rospy_tutorials::AddTwoInts>("/add_two_ints");
    
    rospy_tutorials::AddTwoInts srv;
    srv.request.a = 7;
    srv.request.b = 8;

    if (client.call(srv)) {
        ROS_INFO("[RESULT] Sum = %d", int(srv.response.sum));
    } else {
        ROS_WARN("[ERROR] Service call failed");
    }

}